### DISCUSSION TITLE
Bind to localhost only

### DISCUSSION THREAD

**Discussion by retifrav** [umputun/remark42#1000] [Q&A] [answered]
Is it possible to bind remark42 to loopback/localhost only? At the moment it seems to bind to public interface by default. For example, how it looks like on Linux:

``` sh
$ sudo ss -lntup

tcp  LISTEN  0   151  127.0.0.1:3306  0.0.0.0:*  users:(("mysqld",pid=416,fd=34))
tcp  LISTEN  0  4096          *:8877        *:*  users:(("remark42.linux-",pid=312732,fd=7))
```

So here you can see that I have a MySQL server bound to loopback, so it is not exposed to the internet, and remark42 is bound to public interface. And since I run remark42 (*without Docker*) behind NGINX as reverse-proxy, I would like to bind remark42 to loopback only.

Can this be achieved without modifying the source code and rebuilding it? For example, one would expect an option similar to `--port`, and `--url` looks like the right parameter, but here's how I run remark42:

``` sh
$ remark42.linux-amd64 server --secret=somesecret --url=http://127.0.0.1 --port=8877 --site=some
```

and yet it still binds to all interfaces.

**Comment by umputun** [comment, marked as answer]
not at the moment. the primary configuration is to run it with docker and here it makes sense. But generally, I do agree - the option to limit listening interface is somethat useful

**Comment by umputun** [comment]
see https://github.com/umputun/remark42/pull/1001

**Reply by retifrav** [reply]
Oooooh, this is great. I was just about to add a blocking rule for that port on my server (*just in case*).
Thank you!
